### DISCUSSION TITLE
[質問]: 登録してある日付の背景に色をつけたい

### DISCUSSION THREAD

**Discussion by dreampotter0917** [lit-iphone-course/lit-iphone-course#147] [Question-original]
### エラーの概要

気分が登録してある日付の後ろに丸をつけたく、イメージで表示させようと思ったのですが、なかなかできないです。
また他に良さそうなやり方があったら教えていただきたいです。
結構ずっと悩んでます、、涙

### その他補足

boolでデータの有無は取れます。
日付は年月日で保存してます

### 自分で調べた資料１

https://qiita.com/ayukin/items/03f1e15783f5eed9ef38

### GitHubのURL

https://github.com/dreampotter0917/FeelingApp

### TODOリスト

- [X] GitHubに最新のコードをアップロードしました。
- [X] できるだけ具体的なタイトルを設定しました。

**Comment by nokonoko2057** [comment]
ちょうどスクールで聞かれたのでコメント
上から順に考える順序になってるので、順々に試してもらえると

## やってみること
1. ライブラリのサンプルアプリを動かそう
    - githubの[FSCalendar](https://github.com/WenchaoD/FSCalendar) からファイル一覧をダウンロード
    - FSCalendarExamples.xcworkspace を開く
    - 以下を変更して、実際にコードサンプルコードを動かす
<img width="841" alt="スクリーンショット 2023-03-12 11 40 41" src="https://user-images.githubusercontent.com/12772107/224521243-81b5dddc-c9aa-4805-b5f9-fbd5b56f783b.png">

2. 参考になりそうな部分を見つける 
  - サンプルのアプリを動かすとDelegate Appearanceが参考になりそう。背景の色が変わってる
  - ここのコードが参考になりそう。この関数が呼び出せれば、表示できそう。..... fillDefaultColorFor date: Date)って書いてあるから、日付ごとに呼び出されて、それぞれの色を指定できるっぽい
      - https://github.com/WenchaoD/FSCalendar/blob/2a1bdab38bd4c90caaa4b593e0608c05f4173160/Example-Swift/FSCalendarSwiftExample/DelegateAppearanceViewController.swift#L99-L105
   - どうやったらここを使えるか考える
   - 呼び出し元を探す。ここのViewControllerはFSCalendarDelegateAppearance ってのを継承してて、どうやらそこにこの関数が含まれてるっぽい。
       - https://github.com/WenchaoD/FSCalendar/blob/master/Example-Swift/FSCalendarSwiftExample/DelegateAppearanceViewController.swift#L11
   - ここでDelegateを使うよって宣言してる
       - https://github.com/WenchaoD/FSCalendar/blob/master/Example-Swift/FSCalendarSwiftExample/DelegateAppearanceViewController.swift#L48
   - この辺使えば表示できそう！

**Comment by nokonoko2057** [comment]
@dreampotter0917 コメントしました！

**Comment by txxxxc** [comment]
のこのこさんがめちゃくちゃ丁寧に説明してくれているので、あまり自分から言うこと無いかも、、、笑

一個だけ補足です！

> ここのコードが参考になりそう。この関数が呼び出せれば、表示できそう。..... fillDefaultColorFor date: Date)って書いてあるから、日付ごとに呼び出されて、それぞれの色を指定できるっぽい
https://github.com/WenchaoD/FSCalendar/blob/2a1bdab38bd4c90caaa4b593e0608c05f4173160/Example-Swift/FSCalendarSwiftExample/DelegateAppearanceViewController.swift#L99-L105

`fillDefaultColorFor date: Date) `では該当する日付がDate型で返ってくるのに対して、[らすかるがRealmで保存している日付のデータがString型なので](https://github.com/dreampotter0917/FeelingApp/blob/1cf5c054144846ff62d174408bb9889c7403698a/FeelingApp/FeelingItem.File.swift#L12)、どの日付に色を付けるかif文などで判定する時に結構大変かもです！

RealmはDate型を保存できるので、投稿を保存する時にDate型で日付を保存できるとカレンダーで色を付ける時に判定が簡単かも...!
